core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -4183,79 +4183,86 @@ wasm_interp_call_wasm(WASMModuleInstance *module_inst, WASMExecEnv *exec_env,
 
     wasm_exec_env_set_cur_frame(exec_env, frame);
 
-    if (function->is_import_func) {
+    sigjmp_buf *context = os_create_stack_context();
+    if (!os_is_returning_from_signal(exec_env->handle, context)) {
```

Comment:
how about resource cleanup i mentioned in https://github.com/bytecodealliance/wasm-micro-runtime/issues/1910#issuecomment-1408193240 ?